### PR TITLE
Pie chart labels

### PR DESCRIPTION
**Pull request by alxnddr**
Epic: https://github.com/metabase/metabase/issues/24877

## Changes

Support labels on pie charts:
- Added "Show data labels" setting
- Labels hide when there is not enough space

## How to verify
- Create a pie chart
- Enable "Show data labels" setting
- Ensure labels are visible if pie arcs can fit them in

## Screenshots

Big
Screen Shot 2022-08-30 at 1 05 16 AM

Medium
Screen Shot 2022-08-30 at 1 05 33 AM

Small
Screen Shot 2022-08-30 at 1 05 41 AM

### CODE REVIEW

File: frontend/src/metabase/visualizations/visualizations/PieChart/utils.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+export function getMaxLabelDimension(
+  d3Arc: d3.svg.Arc<d3.svg.arc.Arc>,
+  slice: d3.svg.arc.Arc,
+  index = 0,
```
Comment (WiNloSt) [inline frontend/src/metabase/visualizations/visualizations/PieChart/utils.ts:4]: Are you planning to use this parameter later? But I don't see any usage of this 3rd parameter yet.

File: frontend/src/metabase/visualizations/visualizations/PieChart/utils.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  const arcWidth = outerRadius - innerRadius;
+
+  const arcAngle =
+    d3Arc.startAngle()(slice, index) - d3Arc.endAngle()(slice, index);
+
+  // using law of cosines to calculate the arc length
+  // c = sqrt(a^2 + b^2﹣2*a*b * cos(arcAngle))
+  // where a = b = innerRadius
+
+  const arcLength = Math.sqrt(
```
Comment (WiNloSt) [inline frontend/src/metabase/visualizations/visualizations/PieChart/utils.ts:17]: nit: The name gave me the impression that `arcLength` measure the length of the arc (along the curve = `2*Pi*innerRadius * (arcAngle/(2*Pi))`), but this seems to measure the displacement between the 2 end coordinates that represent the inner arc. I don't know what is the math term for this, but I'm looking for something along the line "the displacement between the 2 points that are the ending of this arc".

**Review by WiNloSt — COMMENTED** [review]
There's also something up with the dashboard with this PR that causes the pie to get cut off. Once this is addressed this PR is then good to go 🎉 

#### Master
Screen Shot 2022-08-30 at 12 38 17 PM

#### This PR
Screen Shot 2022-08-30 at 12 39 24 PM

**Comment by npfitz** [comment]
Pulled down and noticed the same thing @WiNloSt did. Otherwise it looks good to me!

**Comment by alxnddr** [comment]
@npfitz @WiNloSt thanks, I fixed everything you found. Could you have a look again please?

**Review by npfitz — APPROVED** [review]
Positioning looks better!

**Review by WiNloSt — COMMENTED** [review]
Based on my testing maybe we want to reconsider how we decide whether to use white or dark text for pie chart labels.

**Pull request merged**